modules/govuk_containers/manifests/ci_mysql.pp

```diff
@@ -0,0 +1,34 @@
+# == Resource: govuk_containers::ci_mysql
+#
+# Install and run a dockerised MySQL server, intended for CI purposes
+#
+# === Parameters
+#
+# [*version*]
+#   The version of the MySQL Docker image to use
+#
+# [*port*]
+#   The port that the MySQL server will be exposed on
+#
+define govuk_containers::ci_mysql(
+  $ensure = 'present',
+  $version = '8',
+  $port = 33068
+) {
+  ::docker::run { $title:
+    ensure  => $ensure,
+    net     => 'host',
+    expose  => [$port],
```

Comment:
Do you need this to be reachable from the host network or only from within other Docker containers?

Reply:
Thanks for taking a look @sengi. I do need it from the host network. 

I started writing a very different comment but I did some research while trying to backup my understanding, which meant my understanding totally changed. So this comment will now be very different.

So, this currently works and it didn't work when I did the more traditionally ports mapping argument (`-p 33068:3306`) it didn't and I think I now understand why. I understand why this works because we're exposing 33068 and the net argument is set to the host network, this would mean that the Docker container exposes 33068 on the host network.

When I tried to the original port mapping I hit problems with a port conflict on `3306` which conflicted with the host's MySQL installation also using port 3306.

I imagine I can actually make this more idiomatic by not using the host network (which should hopefully resolve the port conflict) and then doing a more traditional docker port mapping. I'll try that out.

Reply:
Yup, removing the `net     => 'host',` meant I could do everything in a more idiomatic way. It's all updated now.

Reply:
LGTM 👍